pkg/handler/resources.go

```diff
@@ -116,10 +116,27 @@ func (h *Handlers) GetNamespaces(ctx context.Context) func(w http.ResponseWriter
 
 func (h *Handlers) getLabelValues(ctx context.Context, cl clients, label string) ([]string, int, error) {
 	if h.PromInventory != nil && h.PromInventory.LabelExists(label) {
-		return prometheus.GetLabelValues(ctx, cl.prom, label, nil)
+		resp, code, err := prometheus.GetLabelValues(ctx, cl.prom, label, nil)
+		if err != nil {
+			if code == http.StatusUnauthorized || code == http.StatusForbidden {
+				// In case this was a prometheus 401 / 403 error, the query is repeated with Loki
+				// This is because multi-tenancy is currently not managed for prom datasource, hence such queries have to go with Loki
+				// Unfortunately we don't know a safe and generic way to pre-flight check if the user will be authorized
+				hlog.Info("Retrying with Loki...")
+				// continuing with loki below
+			} else {
+				return nil, code, fmt.Errorf("error while fetching label %s values from Prometheus: %w", label, err)
+			}
```

Comment:
I don't think we'll need that case as we get https://github.com/netobserv/network-observability-console-plugin/pull/589/files#diff-017f9ad356145a3e9c8d08ec33d1b5e694bc4820f301ad6d5c6470005eaf8a38 changes :thinking: 

The namespace was missing in the query params so we always query the admin prom

Reply:
the error is not only in dev perspective but also in admin perspective, when logged in with a non-admin.. so I think it's still relevant for that case

Reply:
correct, let's keep it then !